src/main/java/ro/msg/learning/shop/controller/ProductCategoryController.java

```diff
@@ -0,0 +1,41 @@
+package ro.msg.learning.shop.controller;
+
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.converter.ProductCategoryConverter;
+import ro.msg.learning.shop.dto.ProductCategoryDto;
+import ro.msg.learning.shop.response.MessageResponse;
+import ro.msg.learning.shop.response.ProductCategoryResponse;
+import ro.msg.learning.shop.service.interfaces.ProductCategoryService;
+
+import java.util.ArrayList;
+
+@RestController
+@RequestMapping("/productCategory")
+public class ProductCategoryController {
+
+
+    @Autowired
+    private ProductCategoryService productCategoryService;
+
+    @Autowired
+    private ProductCategoryConverter productCategoryConverter;
+
+    @GetMapping(value = "")
+    public ProductCategoryResponse getAllProductCategories(){
+        return new ProductCategoryResponse(
+                (ArrayList<ProductCategoryDto>) productCategoryConverter.convertModelToDto(productCategoryService.getAllProductCategories())
```

Comment:
You should not access the Data Model from Controller. So, as it is already suggested in Training material, your service methods should receive as parameters a DTO object and return also a DTO. Please change  this for all Services in the app.